### PR TITLE
add meta data file

### PR DESCRIPTION
**Pull request by Nujanauss**
Unfortunately this pull request is a little longer than I wanted it to be - came across a few bugs - apologies for that!

- If the user cancels file selection they will now be able to continue working rather than throwing an error
- moved device_info check to top because if it is not found, nothing else can proceed, also added check to see if we have device info
- added check to make sure it's possible to write to a given file
- created new meta-data file that includes sampling-rate and adc mask. All other variables that we might want are currently unavailable in the MATLAB API

### CODE REVIEW

File: converter/convertBIN2csv.m [hunk 1]
```diff
@@ -32,6 +37,16 @@
 while read
     packet = parseBtPacket(fid);
     switch packet.type
+        case 'dev_info'
+            if ~exist('device_info', 'var') || isempty(device_info)
+                    disp('No device info discovered')
```
Comment (bmeatayi) [inline converter/convertBIN2csv.m:42]: I am not sure why this message is printed. In my test, when the packet is a device_info packet, the program goes into this case and then the message `No device info discovered` is shown while the current packet is actually a device info packet!

**Comment by bmeatayi** [comment]
The `timestamp`  and `device` columns are removed from the metadata. Maybe it is better to keep those columns always in the file because of analysis pipelines. It would be easier and consistent for the pipeline if the structure of the file is always the same.
The structure of the meta file should look like the meta file generated by explorepy (see the following image).

image

File: converter/convertBIN2csv.m [hunk 2]
```diff
@@ -69,7 +76,4 @@
 end
 
 %% Save data as csv files
-writeCSV(EEG, ORN, Marker, filepath); % Saves two csv files in the same directory as original binary file
-
-
-
+writeCSV(EEG, ORN, Marker, filepath, device_info); % Saves two csv files in the same directory as original binary file
```
Comment (bmeatayi) [inline converter/convertBIN2csv.m:79]: comment is outdated

**Review by tuv-mohamad-atayi — CHANGES_REQUESTED** [review]

File: converter/convertBIN2csv.m (hunk 1 above)
Comment (Nujanauss) [reply converter/convertBIN2csv.m]: Good point! Bad correction by me. Altered now

**Review by tuv-mohamad-atayi — APPROVED** [review]

**Pull request merged**